client/js/slideout.js

```diff
@@ -0,0 +1,102 @@
+"use strict";
+
+const viewport = document.getElementById("viewport");
+const menu = document.getElementById("sidebar");
+
+let touchStartPos = null;
+let touchCurPos = null;
+let touchStartTime = 0;
+let menuWidth = 0;
+let menuIsOpen = false;
+let menuIsMoving = false;
+
+module.exports = {
```

Comment:
Probably silly, but what about:

```js
module.exports = new class SlideoutMenu {
	static enable() { /* ... */ }
	static disable() { /* ... */ }
	status toggle() { /* ... */ }
	status isOpen() { /* ... */ }
};
```

(Would be even nicer if we could create that in a common ancestor file with `new SlideoutMenu()` and pass the ref around, but that's a whooole different arch concern in the current code, hence static functions...)